### PR TITLE
[CDAP-8429] Fix search query with '*'

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-8429

### CODE REVIEW

File: cdap-ui/app/cdap/components/EntityListView/index.js [hunk 1]
```diff
@@ -397,7 +397,7 @@ class EntityListView extends Component {
     };
 
     if (typeof query === 'string' && query.length) {
-      params.query = `${query}*`;
+      params.query = query.charAt(query.length - 1) === '*' ? query : `${query}*`;
```
Comment (bdmogal) [inline cdap-ui/app/cdap/components/EntityListView/index.js:400]: what happens if a user enters ***? Should we even handle it? I don't think so - that's a user error.

File: cdap-ui/app/tracker/controllers/results-ctrl.js [hunk 2]
```diff
@@ -140,8 +140,6 @@ class TrackerResultsController {
     if (params.query === '*') {
       params.sort = 'creation-time desc';
       params.numCursors = 10;
-    } else {
-      params.query = params.query + '*';
```
Comment (bdmogal) [inline cdap-ui/app/tracker/controllers/results-ctrl.js]: we're removing this? Shouldn't it be the same as the CDAP query above?

File: cdap-ui/app/cdap/components/EntityListView/index.js (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/EntityListView/index.js:400]: I don't think so either

File: cdap-ui/app/tracker/controllers/results-ctrl.js (hunk 2 above)
Comment (bdmogal) [reply cdap-ui/app/tracker/controllers/results-ctrl.js]: I think we should keep the behavior of search exactly the same both on the home page and in the metadata search.

**Review by bdmogal — APPROVED** [review]
Verified the fix. Code also LGTM

**Comment by elfenheart** [comment]
Build: http://builds.cask.co/browse/CDAP-DRC5834-3

**Pull request merged**